services/ymax-planner/src/gas-estimation.ts

```diff
@@ -52,19 +59,37 @@ export const makeGasEstimator = ({
     return BigInt(body.trim());
   };
 
-  const getWalletEstimate = async (chainName: AxelarChain) =>
-    queryAxelarGasAPI(
-      AGORIC_CHAIN,
-      chainName,
-      gasLimitEstimates.Wallet,
-      BLD_TOKEN,
-    );
+  const getWalletEstimate = async (
+    chainName: AxelarChain,
+    operationType?: EvmWalletOperationType,
+    protocol?: YieldProtocol,
+  ) => {
+    let gasLimit = walletOperationFallbackGasLimit;
+
+    if (operationType) {
+      const estimates = walletOperationGasLimitEstimates[operationType];
+      if (protocol && estimates?.[protocol]) {
+        gasLimit = estimates[protocol];
+      } else if (estimates) {
+        // If protocol is undefined, pick any value from the estimates object
+        // (useful for DepositForBurn which has the same value for all protocols)
+        const firstValue = Object.values(estimates)[0];
+        if (firstValue !== undefined) {
+          gasLimit = firstValue;
+        }
+      }
+    }
+
+    return queryAxelarGasAPI(AGORIC_CHAIN, chainName, gasLimit, BLD_TOKEN);
```

Comment:
Trying to correct https://github.com/Agoric/agoric-sdk/pull/12076#discussion_r2421069572 with proper multi-line context:
```suggestion
    const estimates = operationType
      ? walletOperationGasLimitEstimates[operationType]
      : {};

    // Absent a protocol-specific gas estimate, pick the first value for this
    // operation type (e.g., DepositForBurn might have the same value for all
    // protocols) or the generic fallback.
    const gasLimit =
      estimates[protocol] ??
      Object.values(estimates)[0] ??
      walletOperationFallbackGasLimit;

    return queryAxelarGasAPI(AGORIC_CHAIN, chainName, gasLimit, BLD_TOKEN);
```